Source/CronosPlayUnreal/Android/SecureStorage.java

```diff
@@ -0,0 +1,89 @@
+package com.cronos.play;
+
+import android.content.Context;
+import android.util.Log;
+import androidx.security.crypto.EncryptedFile;
+import androidx.security.crypto.MasterKey;
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+
+public class SecureStorage {
+
+  public static String readSecureStorage(Context context, String key) {
+    try {
+      MasterKey masterKey =
+          new MasterKey.Builder(context).setKeyScheme(MasterKey.KeyScheme.AES256_GCM).build();
+
+      File file = new File(context.getFilesDir(), key);
+      EncryptedFile encryptedFile =
+          new EncryptedFile.Builder(
+                  context, file, masterKey, EncryptedFile.FileEncryptionScheme.AES256_GCM_HKDF_4KB)
+              .build();
+
+      String result;
+      if (file.exists()) {
+        ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+        InputStream inputStream = encryptedFile.openFileInput();
+        // prepare buffer to read per iteration
+        byte[] buffer = new byte[1024];
+        int len;
+        while ((len = inputStream.read(buffer)) != -1) {
+          outputStream.write(buffer, 0, len);
+        }
+        byte[] data = outputStream.toByteArray();
+        result = new String(data, StandardCharsets.UTF_8);
```

Comment:
should this also close the input / output streams?

Reply:
i'll check